### PR TITLE
Add TGIS test for fetching model from Minio object storage

### PR DESCRIPTION
**Pull request by bdattoma**
Covering RHOAIENG-3490.
PR validation (including regression): `rhods-ci-pr-test/2529` PASS
re-test minio TC after applying Tarun's comment: `rhods-ci-pr-test/2531` PASS

### CODE REVIEW

**Comment by bdattoma** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=red-hat-data-services_ods-ci&pullRequest=1231) **Quality Gate failed**
> Failed conditions  [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=red-hat-data-services_ods-ci&pullRequest=1231&resolved=false&inNewCodePeriod=true)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=red-hat-data-services_ods-ci&pullRequest=1231)

not related to this PR. Will analyse and address in a different one

File: ods_ci/tests/Resources/CLI/Minio.resource [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+${MINIO_RESOURCES_DIRPATH}=    ods_ci/tests/Resources/Files/minio
+${MINIO_POD_FILEPATH}=    ${MINIO_RESOURCES_DIRPATH}/minio.yaml
+${MINIO_SECRET_FILEPATH}=    ${MINIO_RESOURCES_DIRPATH}/minio_secret.yaml
+${MINIO_SA_FILEPATH}=    ${MINIO_RESOURCES_DIRPATH}/minio_serviceaccount.yaml
+
+
+*** Keywords ***
+Deploy MinIO
+    [Documentation]    Deploys MiniIO pod, service and route using the given ${minio_image}
+    [Arguments]    ${minio_image}=quay.io/modh/ods-ci-minio-models:1.0
```
Comment (tarukumar) [inline ods_ci/tests/Resources/CLI/Minio.resource:17]: Use digest for the compatibility for disconnected cluster

File: ods_ci/tests/Resources/CLI/Minio.resource [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    Should Be Equal As Integers    ${rc}    ${0}
+    ${rc}    ${pw}=    Run And Return Rc And Output
+    ...    oc get pod ${podname} -n ${namespace} -o jsonpath='{range .spec.containers[*].env[?(@.name=="MINIO_SECRET_KEY")]}{@.value}{end}'    # robocop: disable
+    Should Be Equal As Integers    ${rc}    ${0}
+    RETURN    ${key}    ${pw}
+
+Remove Minio
+    [Documentation]    Delete the MinIO objects from the cluster
+    Log    message=To Be Implemented
+    # for the time being the test using Minio is deleting the NS.
```
Comment (tarukumar) [inline ods_ci/tests/Resources/CLI/Minio.resource:64]: oc delete all -l app=minio -o name to delete all resouce and then delete the ns

File: ods_ci/tests/Resources/CLI/ModelServing/llm.resource [hunk 3]
```diff
@@ -102,6 +107,11 @@ Create Secret For S3-Like Buckets
     Should Be Equal As Integers    ${rc}    ${0}
     Run Keyword And Ignore Error    Run    oc create -f ${LLM_RESOURCES_DIRPATH}/bucket_sa_filled.yaml -n ${namespace}
     Add Secret To Service Account    sa_name=${sa_name}    secret_name=${name}    namespace=${namespace}
+    IF    ${verify_ssl} == ${FALSE}
+        ${rc}    ${out}=    Run And Return Rc And Output
+        ...    oc annotate secret ${name} -n ${namespace} serving.kserve.io/s3-verifyssl='false' --overwrite
+        Should Be Equal As Integers    ${rc}    ${0}
+    END
```
Comment (tarukumar) [inline ods_ci/tests/Resources/CLI/ModelServing/llm.resource:114]: Do not use this method with ssl support in kserve we need to use below kcs.Also we will have centrlised solution of ssl certs in 2.8 so i think it is good to add this support
https://access.redhat.com/solutions/7053013

File: ods_ci/tests/Resources/Files/minio/minio.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: ${service_name}
```
Comment (tarukumar) [inline ods_ci/tests/Resources/Files/minio/minio.yaml:4]: Add label

File: ods_ci/tests/Resources/CLI/Minio.resource (hunk 1 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/CLI/Minio.resource:17]: ok

File: ods_ci/tests/Resources/CLI/Minio.resource (hunk 2 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/CLI/Minio.resource:64]: I checked and it doesn't block NS deletion.

File: ods_ci/tests/Resources/CLI/ModelServing/llm.resource (hunk 3 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/CLI/ModelServing/llm.resource:114]: this is a temporary workaround. Will add support to SSL

File: ods_ci/tests/Resources/CLI/Minio.resource (hunk 2 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/CLI/Minio.resource]: done

File: ods_ci/tests/Resources/Files/minio/minio.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      port: 9000
+      protocol: TCP
+      targetPort: 9000
+  selector:
+    app: minio
+---
+kind: Route
+apiVersion: route.openshift.io/v1
+metadata:
+  name: ${route_name}
```
[the remark this answers is not in the document]
Comment (bdattoma) [reply ods_ci/tests/Resources/Files/minio/minio.yaml:21]: done

File: ods_ci/tests/Resources/Files/minio/minio.yaml (hunk 4 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/Files/minio/minio.yaml:4]: done

File: ods_ci/tests/Resources/CLI/Minio.resource (hunk 1 above)
Comment (apodhrad) [reply ods_ci/tests/Resources/CLI/Minio.resource]: Hi @bdattoma why do you deploy the Minio? In a disconencted env it is already available.
It is enough just to check test-variables.yaml

File: ods_ci/tests/Resources/CLI/Minio.resource [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    RETURN    ${key}    ${pw}
+
+Get Minio Credentials
+    [Documentation]    Extracts the default MinIO user credentials from the pod env variables
+    [Arguments]    ${namespace}    ${podname}=ods-ci-minio
+    ${rc}    ${key}=    Run And Return Rc And Output
+    ...    oc get pod ${podname} -n ${namespace} -o jsonpath='{range .spec.containers[*].env[?(@.name=="MINIO_ACCESS_KEY")]}{@.value}{end}'    # robocop: disable
+    Should Be Equal As Integers    ${rc}    ${0}
+    ${rc}    ${pw}=    Run And Return Rc And Output
+    ...    oc get pod ${podname} -n ${namespace} -o jsonpath='{range .spec.containers[*].env[?(@.name=="MINIO_SECRET_KEY")]}{@.value}{end}'    # robocop: disable
```
Comment (lugi0) [inline ods_ci/tests/Resources/CLI/Minio.resource:57]: are you sure these values are stored in plaintext and not encoded in e.g. base64?

File: ods_ci/tests/Resources/RHOSi.resource [hunk 7]
```diff
@@ -38,6 +38,8 @@ Resource            Common.robot
 ...                         Login To OCP Using API
 ...                         Generate Thanos Token
 ...                         Set Thanos Credentials Variables
+...                         Generate Minio Random Credentials
```
Comment (lugi0) [inline ods_ci/tests/Resources/RHOSi.resource:41]: if they're random why do we want to hide them?

File: ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/422__model_serving_llm_tgis_object_storages.robot [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+${MODEL_S3_DIR}=    flan-t5-small-hf
+${TEST_NS}=    tgis-storages
+${TGIS_RUNTIME_NAME}=    tgis-runtime
+
+  
+*** Test Cases ***
+Verify User Can Serve And Query A Model From Minio
+    [Documentation]    Basic tests for preparing, deploying and querying a LLM model
+    ...                using Kserve and TGIS runtime fetching models from a MinIO server
+    [Tags]    Tier1    RHOAIENG-3490
```
Comment (lugi0) [inline ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/422__model_serving_llm_tgis_object_storages.robot:24]: isn't it missing the Sanity tag?

**Review by lugi0 — APPROVED** [review]
a couple comments but LGTM

**Review by tarukumar — APPROVED** [review]

File: ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/422__model_serving_llm_tgis_object_storages.robot (hunk 8 above)
Comment (bdattoma) [reply ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/422__model_serving_llm_tgis_object_storages.robot:24]: on purpose cos I'm not sure if all the new tests for serving should fall in sanity which is growing a lot. We could review the quality gates tags all together for model serving

File: ods_ci/tests/Resources/RHOSi.resource (hunk 7 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/RHOSi.resource:41]: makese sense. We can remove it

File: ods_ci/tests/Resources/CLI/Minio.resource (hunk 6 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/CLI/Minio.resource:57]: plain text

File: ods_ci/tests/Resources/CLI/Minio.resource (hunk 1 above)
Comment (bdattoma) [reply ods_ci/tests/Resources/CLI/Minio.resource]: will discuss offline

**Pull request merged**